core/deployment/src/main/java/io/quarkus/deployment/pkg/PackageConfig.java

```diff
@@ -0,0 +1,37 @@
+package io.quarkus.deployment.pkg;
+
+import java.util.List;
+import java.util.Optional;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot
+public class PackageConfig {
```

Comment:
+1 to the idea of putting more of the build config into config objects from the Mojo.